ui/src/utils/generateUiSchema.ts

```diff
@@ -0,0 +1,26 @@
+export function generateUiSchema(
+  schemaProperties: object,
+  uiSchema: Record<string, any> = {},
+  path: string[] = [],
+): Record<string, string> {
+  Object.entries(schemaProperties).forEach(([key, value]) => {
+    if (key === 'properties' && typeof value === 'object' && value !== null) {
+      generateUiSchema(value, uiSchema, path);
+    } else if (typeof value === 'object' && value !== null) {
+      const nestedObject = key === 'properties' ? path : path.concat(key);
+      generateUiSchema(value, uiSchema, nestedObject);
+    } else if (key === 'multiwoven_secret' && value === true) {
+      let current = uiSchema;
+      path.forEach((p, i) => {
```

Comment:
let's use better naming convention here, `p` and `i` does not make that much sense

`schemaPath` and `index`